analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -70,6 +71,61 @@ def parse_analyzer_config(config_dump):
     return re.findall(r'^(\S+):\s+(\S+)$', config_dump, re.MULTILINE)
 
 
+def get_diagtool_bin():
+    """
+    Return full path of diagtool.
+
+    Select clang binary, check for a 'diagtool' binary next to the selected
+    clang binary and return full path of this binary if it exists.
+    """
+    context = analyzer_context.get_context()
+    clang_bin = context.analyzer_binaries.get(ClangSA.ANALYZER_NAME)
+
+    if not clang_bin:
+        return None
+
+    # Resolve symlink.
+    clang_bin = os.path.realpath(clang_bin)
+
+    # Find diagtool next to the clang binary.
+    diagtool_bin = os.path.join(os.path.dirname(clang_bin), 'diagtool')
+    if os.path.exists(diagtool_bin):
+        return diagtool_bin
+
+    LOG.debug("'diagtool' can not be found next to the clang binary (%s)!",
+              clang_bin)
+
+
+def get_warnings(env=None):
+    """
+    Returns list of warning flags by using diagtool.
+    """
+    diagtool_bin = get_diagtool_bin()
+
+    if not diagtool_bin:
+        return []
+
+    try:
+        result = subprocess.check_output(
+            [diagtool_bin, 'tree'],
+            env=env,
+            universal_newlines=True,
+            encoding="utf-8",
+            errors="ignore")
+        return [w[2:] for w in result.split()
+                if w.startswith("-W") and w != "-W"]
+    except subprocess.CalledProcessError as exc:
+        LOG.error("'diagtool' encountered an error while retrieving the "
```

Comment:
Is the second part of the error message correct in this case? 
`If you are using a custom compiled clang, you may have forgotten to build the 'diagtool' target  alongside 'clang' and 'clang-tidy'! `
The diagtool should be there if we catch this exception.

Reply:
Yeah, we should consider rephrasing this message. This exception occurs not only when `diagtool` is not found, but when it is compiled to a different clang which makes `diagtool` crash. This was a discussion in its PR, I'll try to find it an include it as a comment to this line.